### ISSUE TITLE
does securefs support in win10

### ISSUE THREAD

**Issue by zhangjunli177** [netheril96/securefs#107]
Has someone install securefs in win10?
I have installed securefs in win7, there's no problem to use. With the same step, I installed securefs in win10.
the new disk showed up after run the mount command.
While when I tried to create a new dir in the disk, an error  "the disk is not usable" showed up.

googled with key "win10 can't create directory", the answers, modify registry, are not for my case. I can create a new dir in my desktop, but just not in the new disk.

Here're the version info:
PS C:\Users\zhang\Desktop\securefs-win64> .\securefs.exe version
securefs 0.11.1
Crypto++ 8.2
WinFsp 1.6
utf8proc 2.5.0

Hardware features available:
SSE2: true
SSE3: true
SSE4.1: true
SSE4.2: true
AES-NI: true
CLMUL: true
SHA: false

**Comment by netheril96** [comment]
I use `securefs` constantly on Win10. In fact, I never test `securefs` on Win7. So, Win10 is actually the best supported Windows version.

As for your issue, I need more details. What are the logs from `securefs`? Is it a fresh Win10 installation? Have you updated to the latest Win10?

**Comment by zhangjunli177** [comment]
Thanks for your reply. I have updated the latest win10. Here's the log I failed to create a folder.
[win10.log](https://github.com/netheril96/securefs/files/4750401/win10.log)
<img width="866" alt="front" src="https://user-images.githubusercontent.com/7940414/84115994-96656b00-aa61-11ea-9716-5ba1f3718516.png">

**Comment by zhangjunli177** [comment]
I guess this maybe a windows folder permission issue. Sorry for can't show english in the picture as my OS drop doesn't support multi-language. While the icon should be directive
Here's the screen shot when I create a new folder in C:
<img width="960" alt="new-c" src="https://user-images.githubusercontent.com/7940414/84118223-44264900-aa65-11ea-826e-a5d7b2ddffac.png">

Here's the screen shot when I create a new folder in Z:
<img width="960" alt="new-z" src="https://user-images.githubusercontent.com/7940414/84118284-5a340980-aa65-11ea-9fca-545eee1b0684.png">

When I check the permission on Z, full control, read, write, et etc are unchecked. And only an specical permission is checked. And I'm not allowed to check the full control options.

**Comment by netheril96** [comment]
> Sorry for can't show english

Don't worry, I'm Chinese too. Although I prefer this conversation to continue in English, so that future readers might benefit from the solution.

> When I check the permission on Z, full control, read, write, et etc are unchecked. And only an specical permission is checked. And I'm not allowed to check the full control options.

`securefs` mount points have the same permissions as the underlying folders. Can you create new folders on C:\JJZ directly? If not, that will be the cause.

Regardless of whether that succeeds or fails, can you try creating and mounting a folder from your desktop or home folder?

**Comment by netheril96** [comment]
By the way, the screenshot has displayed the master key in plaintext. Remember to delete C:\JJZ later.

**Comment by zhangjunli177** [comment]
After I mount C:\JJZ to Z:, I can create the dir in C:\JJZ w/o any issue.
I have tried to mount a folder in desktop, still the same result. Here's the error when create the folder.
<img width="418" alt="createfolder" src="https://user-images.githubusercontent.com/7940414/84169983-d94c3080-aaab-11ea-97e4-334fe6405e92.png">

**Comment by netheril96** [comment]
Can you try if [mounting memfs](http://www.secfs.net/winfsp/rel/#launch-winfsp-from-the-windows-explorer) works? I need to know if the issue is specific to `securefs`.

**Comment by zhangjunli177** [comment]
mounting memfs works for me. See the picture below.
Found a post yesterday for the virtual disk issue in vmware win10, the solution is restart explorer with admin permission. I tried that, but didn't fix my issue.
https://www.itsk.com/thread-395102-1-1.html

<img width="606" alt="memfs" src="https://user-images.githubusercontent.com/7940414/84225385-915ff480-ab11-11ea-8d93-1dd8762d6058.png">

**Comment by zhangjunli177** [comment]
I also tried to mount a local directory with winfsp, there's no issue to create a dir in the new disk.

**Comment by netheril96** [comment]
It is really difficult for me to debug because I cannot reproduce the issue.

Some more ideas:
* Try opening a command prompt in administrator mode, and run securefs there
* Try mounting to a different drive point, such as X:
* Try mounting to a folder, such as D:\YYZZZ
* Try creating directory in the mount point on the command line

**Comment by zhangjunli177** [comment]
Actually I tried them before I opened the ticket. I think this should be a system configuration issue instand of coding defect. I'll continue to trace the issue, and will post the solution if I find.
Thanks anyway for your help.

**Comment by bryanlyon** [comment]
I haven't had this problem on most of my systems, but just installed on a new Windows device.  Getting this problem (No permissions ONLY in mounted systems.  This is happening on a fully updated WIndows system trying the last 3 "releases" of winfsp.  1.7.20172 shows 

"[Window Title]
Location is not available

[Content]
X:\ is unavailable. If the location is on this PC, make sure the device or drive is connected or the disc is inserted, and then try again. If the location is on a network, make sure you’re connected to the network or Internet, and then try again. If the location still can’t be found, it might have been moved or deleted.

[OK]"

While 1.8.20186 and 1.8.20221 both show "Permission Denied" errors.  (Sorry I don't have the exact message, I can get it if you want but it requires restarting so can't get this error AND the 1.7 error at the same time.

This only happens on this one system (which unfortunately is the one I need it to work on at the moment), another system with the 1.8.20186 version of winfsp works perfectly.

All versions load the folder correctly but cannot write.  As nothing is shown in the securefs log, I'm guessing it's actually a problem with winfsp *OR* with how securefs is interfacing with it that has changed in some update.

**Comment by bryanlyon** [comment]
Found that with --trace there are actually calls to secureFS showing in the log, but no errors.  That said, I did find something interesting.  By manually creating folders matching paths from the trace log, I was still unable to write any data to the securefs, but I *WAS* able to delete those folders.  This means that it's not at all a permissions issue underneath SecureFS but something internal between SecureFS-WinFSP-And some instances of Windows 10.
